src/Sfa.Tl.Matching.Application.IntegrationTests/MatchingServiceReport/Builders/OpportunityBuilder.cs

```diff
@@ -0,0 +1,252 @@
+                OpportunityType = OpportunityType.Referral.ToString(),
+
+                RouteId = 1,
+                SearchResultProviderCount = 1,
+                CreatedBy = "Sfa.Tl.Matching.Application.IntegrationTests",
+                Referral = providerVenueId.Select(pv => new Referral
+                {
+                    DistanceFromEmployer = 1.1m,
+                    ProviderVenueId = pv,
+                    CreatedBy = "Sfa.Tl.Matching.Application.IntegrationTests"
+                }).ToList()
+            };
+        }
+
+        public OpportunityItem CreateProvisionGapOpportunityItem(bool isSaved, bool isCompleted)
+        {
+            return new OpportunityItem
+            {
+                Town = "test",
+                JobRole = "test",
+                Postcode = "test",
+
+                Placements = 1,
+                PlacementsKnown = true,
+
+                IsSaved = isSaved,
+                IsCompleted = isCompleted,
+                IsSelectedForReferral = true,
+                OpportunityType = OpportunityType.ProvisionGap.ToString(),
+
+                RouteId = 1,
+                SearchResultProviderCount = 1,
+                CreatedBy = "Sfa.Tl.Matching.Application.IntegrationTests",
+                ProvisionGap = new List<ProvisionGap>
+                {
+                    new ProvisionGap
+                    {
+                        HadBadExperience = false,
+                        NoSuitableStudent = false,
+                        ProvidersTooFarAway = false,
+                        CreatedBy = "Sfa.Tl.Matching.Application.IntegrationTests",
+                    }
+                }
+            };
+        }
+
+        public void ClearData()
+        {
+            var referral = _context.Referral.Where(r => r.CreatedBy == "Sfa.Tl.Matching.Application.IntegrationTests").ToList();
+            if (!referral.IsNullOrEmpty()) _context.Referral.RemoveRange(referral);
+
+            var provisionGap = _context.ProvisionGap.Where(r => r.CreatedBy == "Sfa.Tl.Matching.Application.IntegrationTests").ToList();
+            if (!provisionGap.IsNullOrEmpty()) _context.ProvisionGap.RemoveRange(provisionGap);
+
+            var oppItems = _context.OpportunityItem.Where(o => o.CreatedBy == "Sfa.Tl.Matching.Application.IntegrationTests").ToList();
+            if (!oppItems.IsNullOrEmpty()) _context.OpportunityItem.RemoveRange(oppItems);
+
+            var opp = _context.Opportunity.Where(o => o.CreatedBy == "Sfa.Tl.Matching.Application.IntegrationTests").ToList();
+            if (!opp.IsNullOrEmpty()) _context.Opportunity.RemoveRange(opp);
+
+            _context.SaveChanges();
+        }
+    }
+
+    public class ProviderBuilder
+    {
+        private readonly MatchingDbContext _context;
+
+        public ProviderBuilder(MatchingDbContext context)
+        {
+            _context = context;
+        }
+
+        public Provider CreaeProvider(int venueCount = 1)
```

Comment:
CreateProvider, please?

Reply:
Should there be 3 files for the 3 classes in this file?